images/devtools-kubernetes-v1beta1/context/magefiles/argocd.go

```diff
@@ -0,0 +1,93 @@
+//go:build mage
+// +build mage
+
+package main
+
+import (
+	"fmt"
+	"github.com/magefile/mage/sh" // sh contains helpful utility functions, like RunV
+	"gopkg.in/yaml.v3"
+	"os"
+	"strings"
+)
+
+type ArgoCDAppHelm struct {
+	ReleaseName string   `yaml:"releaseName"`
+	ValueFiles  []string `yaml:"valueFiles"`
+}
+
+type ArgoCDAppSource struct {
+	Helm ArgoCDAppHelm `yaml:"helm"`
+	Path string        `yaml:"path"`
+}
+
+type ArgoCDAppSpec struct {
+	Source ArgoCDAppSource `yaml:"source"`
+}
+
+type ArgoCDAppMetadata struct {
+	Name string `yaml:"name"`
+}
+
+type ArgoCDApp struct {
+	Spec     ArgoCDAppSpec     `yaml:"spec"`
+	Metadata ArgoCDAppMetadata `yaml:"metadata"`
+}
+
+func getArgoCDDeployments(repoURL string) ([]ArgoCDApp, error) {
+	var argoCDAppList []ArgoCDApp
+	env := map[string]string{}
+	if token, ok := os.LookupEnv("ARGOCD_API_TOKEN"); ok {
+		env["ARGOCD_API_TOKEN"] = token
+	}
+	appYaml, err := sh.OutputWith(env, "argocd", "--grpc-web", "app", "list", "-r", repoURL, "-o", "yaml")
```

Comment:
I get the following error when running validate, I have a valid catalogue file.

```
FATA[0000] Argo CD server address unspecified           
Failed to score files: open : no such file or directory
kube-score returned exit code: 1
 Output:
  Error:
 running "kube-score score " failed with exit code 1
Error: running "kube-score score " failed with exit code 1
```

I assume we need to have a working / authenticated argocd setup before running the devtool (hence the `.argocd` mount). We'll likely need to document this as it wasn't immediately obvious to me why it was failing.

Reply:
implemented a check and feedback on auth failure.

Reply:
mmm that is actually not pointing to the lines..